src/backend/opencl/kernel/iops.cl

```diff
@@ -23,8 +30,9 @@ void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 #ifdef MAX_OP
 void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 {
-    if ((sabs(lhs[0]) < sabs(rhs)) ||
-        (sabs(lhs[0]) == sabs(rhs) && *lidx > ridx)) {
+    if (!(IS_NAN(lhs[0])) &&
```

Comment:
Switch order of isNan operation. See other comments for info.

Reply:
👍